### PR TITLE
io_uring: Merge backported io_uring support into the master branch

### PR DESCRIPTION
**Pull request by temerkhanov**
- Backport io_uring from kernel 5.9 to new-kernel
- Enable io_uring support in xen-tools
- Enable io_uring support in EVE
- Add misc kernel fixes

### CODE REVIEW

**Review by rvs — CHANGES_REQUESTED** [review]
Hey @temerkhanov -- thanks for putting it together. Please answer the questions about the individual patches re-jiggering.

On top of that, I would still like this to soak much more before we make any of this official. To this end:
  1. I see that you updated the new-kernel which is exactly the right thing to do
  2. I am obviously uneasy when it comes to `thread -> io_uring` change in kvm.go. The problem is that if we merge your PR as-is our master will break (since the default build is kernel and not new-kernel). So perhaps, one way to have our cake and eat it too is for kvm.go to test for the kernel version and if it is 5.4.72 or higher default to io_uring and if it isn't default to threads. I am open to other ideas as well.

**Comment by rvs** [comment]
Oh, and one more question: can we kill storage branch now or do you still want to use it for something @temerkhanov ?

File: pkg/new-kernel/patches-5.4.x/0001-swiotlb-xen-use-vmalloc_to_page-on-vmalloc-virt-addr.patch [hunk 1]
```diff
@@ -1,44 +0,0 @@
-From f5f4de59377afdd9a80e0f47df109eacf427c203 Mon Sep 17 00:00:00 2001
```
[the remark this answers is not in the document]
Comment (temerkhanov) [reply pkg/new-kernel/patches-5.4.x/0001-swiotlb-xen-use-vmalloc_to_page-on-vmalloc-virt-addr.patch:1]: Yes, an updated version of this patch is included in 5.4.72

File: pkg/new-kernel/patches-5.4.x/0011-adding-page-coherent-patch.patch [hunk 2]
```diff
@@ -1,39 +0,0 @@
-From 1f1182bf84ec7a6d1ec74c56fccd66ff013077af Mon Sep 17 00:00:00 2001
```
[the remark this answers is not in the document]
Comment (temerkhanov) [reply pkg/new-kernel/patches-5.4.x/0011-adding-page-coherent-patch.patch:1]: Yes, an updated version of this patch is included in 5.4.72

File: pkg/new-kernel/patches-5.4.x/0015-Use-chip-data-for-storing-per-IRQ-XEN-data-pointer.patch [hunk 3]
```diff
@@ -1,103 +0,0 @@
-From: Thomas Gleixner <tglx@linutronix.de>
```
[the remark this answers is not in the document]
Comment (temerkhanov) [reply pkg/new-kernel/patches-5.4.x/0015-Use-chip-data-for-storing-per-IRQ-XEN-data-pointer.patch:1]: Yes, this patch is included in 5.4.72

File: pkg/pillar/hypervisor/kvm.go [hunk 4]
```diff
@@ -441,8 +443,29 @@ func (ctx kvmContext) CreateDomConfig(domainName string, config types.DomainConf
...
+		major, _ = strconv.Atoi(osver[0])
+	}
+	if len(osver) >= 2 {
+		minor, _ = strconv.Atoi(osver[1])
+	}
+	if len(osver) >= 3 {
+		patch, _ = strconv.Atoi(osver[2])
+	}
+
+	if major >= 5 && minor >= 4 && patch >= 72 {
```
Comment (rvs) [inline pkg/pillar/hypervisor/kvm.go:465]: Two suggestions here @temerkhanov 
   1. could you please turn 5.4.72 into something like a `const minUringKernelVersion = [5, 4, 72]` at the top of this file (feel free to come up with better name)?
   2. let's make the comparison real (jut to make sure that should we upgrade to 5.10.0 this check will still pass)?

**Review by rvs — APPROVED** [review]
Once small nit -- LGTM otherwise

**Review by rvs — APPROVED** [review]
LGTM

**Pull request merged**